pkg/cmd/roachtest/roachtestutil/task/manager_test.go

```diff
@@ -135,6 +135,53 @@ func TestTerminate(t *testing.T) {
 	require.Equal(t, uint32(numTasks), counter.Load())
 }
 
+func TestGroups(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+	m := NewManager(ctx, nilLogger())
+
+	numTasks := 10
+	g := m.NewGroup()
+	channels := make([]chan struct{}, 10)
+
+	// Start tasks.
+	for i := 0; i < numTasks; i++ {
+		channels[i] = make(chan struct{})
+		g.Go(func(ctx context.Context, l *logger.Logger) error {
+			<-channels[i]
+			return nil
+		})
+	}
+
+	// Start a goroutine that waits for all tasks in the group to complete.
+	done := make(chan struct{})
+	go func() {
+		g.Wait()
+		close(done)
+	}()
+
+	// Close channel one by one to complete all tasks, and ensure the group is not
+	// done yet.
+	for i := 0; i < numTasks; i++ {
+		select {
+		case <-done:
+			t.Fatal("group should not be done yet")
+		default:
+		}
+		close(channels[i])
+		<-m.CompletedEvents()
+	}
+
+	select {
+	case <-done:
+	case <-time.After(30 * time.Second):
```

Comment:
I'm curious if this is good practice to have a timeout in a test, or rather wait indefinitely and let a "global test timeout" happen?

Reply:
I'm not sure how stress/race interacts with time, but it does feel like this would break under that.